<robot name="robot">
    <link name="base_link">
        <visual>
            <origin rpy="1.57 0 1.57" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://PID/meshs/side.stl" scale="0.1 0.1 0.1"/>
            </geometry>
        </visual>
    </link>
    <link name="dc_roller">
        <visual>
            <origin rpy="1.57 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://PID/meshs/dc_roll.stl" scale="0.1 0.1 0.1"/>
            </geometry>
        </visual>
    </link>
    <link name="loadcell_plate">
        <visual>
            <origin rpy="1.57 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://PID/meshs/loadcell_plate.stl" scale="0.1 0.1 0.1"/>
            </geometry>
        </visual>
    </link>
    <link name="loadcell">
        <visual>
            <origin rpy="1.57 0 1.57" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://PID/meshs/loadcell.stl" scale="0.1 0.1 0.1"/>
            </geometry>
        </visual>
    </link>
    <link name="loadcell_pressure">
        <visual>
            <origin rpy="1.57 0 1.57" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://PID/meshs/loadcell_pressure.stl" scale="0.1 0.1 0.1"/>
            </geometry>
        </visual>
    </link>
    <link name="servo_support">
        <visual>
            <origin rpy="1.57 0 0" xyz="0 0 0" />
            <geometry>
                <mesh filename="package://PID/meshs/servo_support.stl" scale="0.1 0.1 0.1"/>
            </geometry>
        </visual>
    </link>
    <link name="servo_arm">
        <visual>
            <origin rpy="3.14 0 1.57" xyz="0 0 1" />
            <geometry>
                <mesh filename="package://PID/meshs/servo_arm.stl" scale="0.1 0.1 0.1"/>
            </geometry>
        </visual>
    </link>


    <joint name="base_to_dc" type="fixed">
        <parent link="base_link" />
        <child link="dc_roller" />
        <origin xyz="25 -8 3" rpy="0 0 0" />
        <axis xyz="0 1 0" />
        <!-- <limit lower="-3.14" upper="3.14" effort="20" velocity="5" /> -->
    </joint>


    <joint name="j2" type="fixed">
        <parent link="base_link" />
        <child link="loadcell_plate" />
        <origin xyz="5 -3 0" rpy="0 0 0" />
        <axis xyz="0 0 0" />
        <!-- <limit lower="-3.14" upper="3.14" effort="20" velocity="5" /> -->
    </joint>

    <joint name="j3" type="fixed">
        <parent link="loadcell_plate" />
        <child link="loadcell" />
        <origin xyz="0 1 3" rpy="0 0 0" />
        <axis xyz="0 0 0" />
        <!-- <limit lower="-3.14" upper="3.14" effort="20" velocity="5" /> -->
    </joint>

    <joint name="j4" type="fixed">
        <parent link="loadcell_plate" />
        <child link="loadcell_pressure" />
        <origin xyz="0 2.6 4" rpy="0 0 0" />
        <axis xyz="0 0 0" />
        <!-- <limit lower="-3.14" upper="3.14" effort="20" velocity="5" /> -->
    </joint>

    <joint name="j5" type="fixed">
        <parent link="base_link" />
        <child link="servo_support" />
        <origin xyz="15 -4 0" rpy="0 0 0" />
        <axis xyz="0 0 0" />
        <!-- <limit lower="-3.14" upper="3.14" effort="20" velocity="5" /> -->
    </joint>

    <joint name="servo_to_arm" type="revolute">
        <parent link="servo_support" />
        <child link="servo_arm" />
        <origin xyz="0 3 5" rpy="0 0 0" />
        <axis xyz="0 1 0" />
        <limit lower="-0.8" upper="2.35" effort="20" velocity="5" />
    </joint>

</robot>